notebooks/0110_CAMS.py

```diff
@@ -0,0 +1,275 @@
+        fileName = "CAMS-GLOB-ANT_Glb_0.1x0.1_anthro_" + gas + "_v6.2_yearly_" + str(y) + ".nc"
+        fullName = folderName + "/" + fileName
+        print(fullName)
+        cams_data_file = Path(cams_data_folder, fullName)
+        emissions = xr.open_dataset(cams_data_file)
+        # compute world total for ships
+        world_emissions_shp_yr = emissions["shp"].sum(["lat", "lon"])
+        # coarsen to the resolution of the iso3 mask
+        # we need to go from a 0.1x0.1 grid to a 0.5x0.5 grid
+        emissions = emissions.coarsen(lat=5, lon=5, boundary="trim").sum()
+        print("coarsened")
+        # align grids by interpolation (is this ok?)
+        emissions = emissions.interp(lon=idxr.lon, lat=idxr.lat, method="linear")
+        print("interpolated")
+        for k, reg in enumerate(iso3_list):
+            country_emissions_reg = (emissions * idxr.sel(iso=reg)).sum(["lat", "lon"])
+            if k == 0:
+                # if the array does not exist, create it
+                country_emissions_yr = country_emissions_reg
+            else:
+                country_emissions_yr = xr.concat([country_emissions_yr, country_emissions_reg], dim="iso")
+        if j == 0:
+            country_emissions_gas = country_emissions_yr
+            world_emissions_shp_gas = world_emissions_shp_yr
+        else:
+            country_emissions_gas = xr.concat([country_emissions_gas, country_emissions_yr], dim="time")
+            world_emissions_shp_gas = xr.concat([world_emissions_shp_gas, world_emissions_shp_yr], dim="time")
+        del country_emissions_yr
+        del world_emissions_shp_yr
+        # print("time for concatenation ", perf_counter()-t0)
+    # turn "nmvocs" to "nmvoc" for consistency
+    if gas == "nmvocs":
+        country_emissions_gas = country_emissions_gas.assign_coords(gas="NMVOC")
+        world_emissions_shp_gas = world_emissions_shp_gas.assign_coords(gas="NMVOC")
+    elif gas == "nox":
+        country_emissions_gas = country_emissions_gas.assign_coords(gas="NOx")
+        world_emissions_shp_gas = world_emissions_shp_gas.assign_coords(gas="NOx")
+    else:
+        # default: just convert to uppercase
+        country_emissions_gas = country_emissions_gas.assign_coords(gas=gas.upper())
+        world_emissions_shp_gas = world_emissions_shp_gas.assign_coords(gas=gas.upper())
+    print("assigned")
+    if i == 0:
+        country_emissions = country_emissions_gas
+        world_emissions_shp = world_emissions_shp_gas
+    else:
+        country_emissions = xr.concat([country_emissions, country_emissions_gas], dim="gas")
+        world_emissions_shp = xr.concat([world_emissions_shp, world_emissions_shp_gas], dim="gas")
+    del country_emissions_gas
+    del world_emissions_shp_gas
+
+# %% [markdown]
+# Now the CAMS-GLOB-AIR data is imported.
+#
+# **TODO**: the sum sector of CAMS-GLOB-ANT does not contain air and so could be misleading, should we remove it?
```

Comment:
Sounds like a sensible thing to do. Also, since CAMS includes agricultural waste burning which CEDS does not, the dataset sums do not map one-to-one.